util/log.py

```diff
@@ -22,3 +32,26 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return outer
+
+
+@contextmanager
+def elapsed_time_logging(
+    *, log_method: Callable, message_prefix: str = None, skip_start=False
+):
+    """Context manager for logging elapsed time.
+
+    :param log_method: Callable to be used to log the message(s).
+    :param message_prefix: Optional string to be prepended to the emitted log records.
+    :param skip_start: Boolean indicating whether to skip the starting message.
+    """
+
+    prefix = f"{message_prefix}: " if message_prefix else ""
+    if not skip_start:
+        log_method(f"{prefix}Starting...")
```

Comment:
If we have a prefix, then it probably makes sense to use a lower case of `Starting...`